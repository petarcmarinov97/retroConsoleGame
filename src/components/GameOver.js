import React, { Component } from 'react';

export default class GameOver extends Component {
    render() {
        const gameOverMsg = "The Game is Over!\n Do you want to start a new game ?";
        const { defaultStyle } = this.props;

        return <div id="gameOver" style={defaultStyle}>
            <span>{gameOverMsg}</span>
            <div>
                <button id="Yes">Yes</button>
                <button id="No">No</button>
            </div>
        </div>
    }
}
