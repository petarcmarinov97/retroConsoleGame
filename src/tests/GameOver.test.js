import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import GameOver from '../components/GameOver';

describe('GameOver tests', () => {

    let wrapper;
    const defaultStyle = {
        display: "none"
    }

    beforeEach(() => {
        wrapper = shallow(
            <GameOver defaultStyle={defaultStyle} />);
    })

    describe('Snapshots', () => {
        it("should render correctly", () => {
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        })
    })
})