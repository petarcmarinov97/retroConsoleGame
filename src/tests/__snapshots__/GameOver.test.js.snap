// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver tests Snapshots should render correctly 1`] = `
<div
  id="gameOver"
  style={
    Object {
      "display": "none",
    }
  }
>
  <span>
    The Game is Over!
 Do you want to start a new game ?
  </span>
  <div>
    <button
      id="Yes"
    >
      Yes
    </button>
    <button
      id="No"
    >
      No
    </button>
  </div>
</div>
`;
